src/compressed_tensors/compressors/model_compressor.py

```diff
@@ -172,24 +176,28 @@ def from_pretrained_model(
         )
 
     @staticmethod
-    def parse_sparsity_config(compression_config: Dict) -> Union[Dict, None]:
+    def parse_sparsity_config(
+        compression_config: Union[Dict[str, Any], "CompressedTensorsConfig"]
+    ) -> Union[Dict[str, Any], None]:
+        """
+        Parse sparsity config from quantization/compression config. Sparsity
+        config is nested inside q/c config
+
+        :param compression_config: quantization/compression config
+        :return: sparsity config
+        """
         if compression_config is None:
             return None
-        if SPARSITY_CONFIG_NAME not in compression_config:
```

Comment:
This line was causing the bug where sparsity config is None in the HF quantizer case